packages/dev/core/src/Misc/textureTools.ts

```diff
@@ -230,6 +231,179 @@ export function FromHalfFloat(value: number): number {
+
+    if (rtt.renderTarget && internalTexture) {
+        const samplingMode = internalTexture.samplingMode;
+        if (lod !== 0) {
+            texture.updateSamplingMode(Texture.NEAREST_NEAREST_MIPNEAREST);
+        } else {
+            texture.updateSamplingMode(Texture.NEAREST_NEAREST);
+        }
+
+        scene.postProcessManager.directRender([lodPostProcess], rtt.renderTarget, true);
+        texture.updateSamplingMode(samplingMode);
+
+        // Read the contents of the framebuffer
+        const numberOfChannelsByLine = width * 4;
+        const halfHeight = height / 2;
+
+        //Reading datas from WebGL
+        const bufferView = await engine.readPixels(0, 0, width, height);
+        const data = new Uint8Array(bufferView.buffer, 0, bufferView.byteLength);
+
+        if (!channels.R || !channels.G || !channels.B || !channels.A) {
+            for (let i = 0; i < width * height * 4; i += 4) {
+                // If alpha is the only channel, just display alpha across all channels
+                if (channels.A && !channels.R && !channels.G && !channels.B) {
+                    data[i] = data[i + 3];
+                    data[i + 1] = data[i + 3];
+                    data[i + 2] = data[i + 3];
+                    data[i + 3] = 255;
+                    continue;
+                }
+                let r = data[i],
+                    g = data[i + 1],
+                    b = data[i + 2],
+                    a = data[i + 3];
+                // If alpha is not visible, make everything 100% alpha
+                if (!channels.A) {
+                    a = 255;
+                }
+                // If only one color channel is selected, map both colors to it. If two are selected, the unused one gets set to 0
+                if (!channels.R) {
+                    if (channels.G && !channels.B) {
+                        r = g;
+                    } else if (channels.B && !channels.G) {
+                        r = b;
+                    } else {
+                        r = 0;
+                    }
+                }
+                if (!channels.G) {
+                    if (channels.R && !channels.B) {
+                        g = r;
+                    } else if (channels.B && !channels.R) {
+                        g = b;
+                    } else {
+                        g = 0;
+                    }
+                }
+                if (!channels.B) {
+                    if (channels.R && !channels.G) {
+                        b = r;
+                    } else if (channels.G && !channels.R) {
+                        b = g;
+                    } else {
+                        b = 0;
+                    }
+                }
+                data[i] = r;
+                data[i + 1] = g;
+                data[i + 2] = b;
+                data[i + 3] = a;
+            }
+        }
+
+        //To flip image on Y axis.
+        if ((texture as Texture).invertY || texture.isCube) {

```

Comment:
Same thing here, I think the Y inversion / cube management is intended to display the texture the right way round in the inspector. But for an exporter, we probably don't want to modify the data. So, maybe add a new parameter to the function to handle this case?

Reply:
Yup this part should stay in the inspector only.

Reply:
Ok, I moved the channel filtering and flip back to the inspector.